api/files.py

```diff
@@ -0,0 +1,140 @@
+import datetime
+import hashlib
+import logging
+import shutil
+import json
+import os
+
+from . import util
+
+log = logging.getLogger('scitran.api')
+
+
+class FileRequest(object):
+    """This class provides and interface for file uploads.
+    To perform an upload the client of the class should follow these steps:
+
+    1) initialize the request
+    2) save a temporary file
+    3) check identical
+    4) move the temporary file to its destination
+
+    The operations could be safely interleaved with other actions like permission checks or database updates.
+    """
+
+    def __init__(self, client_addr, filename, body, received_md5, metadata, tags):
```

Comment:
We discussed this at the time with Kevin, and we decided to use MD5 as some cloud services (eg Amazon S3) supports it and it is not very computationally expensive specially on the browser.

Now there is a discussion going on with @gsfr and @josschne about not supporting any hashing on file transport at all.
Maybe we can continue it on a github issue. Let me know what you think about.